tnoodle-ui/src/App.js

```diff
@@ -2,12 +2,62 @@ import React, { Component } from "react";
 import EntryInterface from "./components/EntryInterface";
 import EventPickerTable from "./components/EventPickerTable";
 import SideBar from "./components/SideBar";
-import FlashMessage from "./components/FlashMessage";
 import VersionInfo from "./components/VersionInfo";
 import "bootstrap/dist/css/bootstrap.css";
 import "./App.css";
+import fetchIntercept from "fetch-intercept";
 
 class App extends Component {
+    constructor() {
+        super();
+
+        const that = this;
+
+        // http interceptor
+        fetchIntercept.register({
+            response: function (response) {
+                if (!response.ok) {
+                    that.handleHttpError(response);
+                }
+                return response;
+            },
+
+            responseError: function (error) {
+                that.handleHttpError(error);
+                return Promise.reject(error);
+            },
+        });
+
+        this.state = { errorMessage: "" };
+    }
+
+    handleHttpError = (error) => {
+        let message = error.statusText || JSON.stringify(error);
+        this.updateErrorMessage(message);
+    };
+
+    updateErrorMessage = (errorMessage) => {
+        // Clear the message after some seconds
+        setTimeout(() => {
+            this.setState({ ...this.state, errorMessage: "" });
+        }, 1000 * 10);
```

Comment:
Magic number calculation. Better create a constant "errorTimoutInSeconds" somewhere and reference that here

Reply:
Updated, thanks! I just changed the "timeout" part because I think this would make it confusing somehow.